File: TypeScript/1. Two Sum.ts
function twoSum(nums: number[], target: number): number[] {
    let hash: { [key: number]: number } = {};
    
    for (let i = 0; i < nums.length; i++) {
        let subtracted = target - nums[i];

        if (subtracted in hash) 
            return [hash[subtracted], i];

        hash[nums[i]] = i;
    }
    return [];
};

console.log(twoSum([2,7,11,15], 9));

File: Go/1. Two Sum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	var hashTable = make(map[int]int)
	for i := 0; i < len(nums); i++ {
		subtracted := target - nums[i]
		if val, ok := hashTable[subtracted]; ok {
			return []int{val, i}
		}
		hashTable[nums[i]] = i
	}
	return []int{0, 0}
}

func main() {
	nums := []int{2, 7, 11, 15}
	target := 9
	fmt.Println(twoSum(nums, target))
}
